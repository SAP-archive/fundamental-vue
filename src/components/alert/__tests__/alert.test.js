import { TransitionGroupStub, TransitionStub, mount } from '@vue/test-utils'
import { Alert } from '../'

describe('Alert', () => {
  it('renders correctly', async () => {
    const warningAlert = await mount(Alert, {
      propsData: {
        type: 'warning',
        dismissible: true,
        id: 'warningAlert',
        uid: 'alert'
      }
    })
    const errorAlert = await mount(Alert, {
      propsData: { type: 'error', dismissible: false, id: 'errorAlert' }
    })
    expect(warningAlert.element).toMatchSnapshot()
    expect(errorAlert.element).toMatchSnapshot()
    expect(warningAlert.classes('fd-alert--warning')).toBe(true)
    expect(errorAlert.classes('fd-alert--dismissible')).toBe(false)
  })
  it('renders default slot when passed', () => {
    const dummySlot = 'Slot text'
    const wrapper = mount(Alert, {
      slots: {
        default: dummySlot
      }
    })
    expect(wrapper.text()).toMatch(dummySlot)
  })
  it('click on close hides the alert', async () => {
    const alert = mount(
      Alert,
      {
        propsData: {
          dismissible: true
        }
      },
      {
        stubs: {
          transition: TransitionStub,
          'transition-group': TransitionGroupStub
        }
      }
    )
    expect(alert.isVisible()).toBe(true)
    const closeButton = alert.find('button')
    expect(alert.isVisible()).toBe(true)
    await closeButton.trigger('click')
    expect(alert.isVisible()).toBe(false)
  })
})
