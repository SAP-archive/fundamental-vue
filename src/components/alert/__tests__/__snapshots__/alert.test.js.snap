// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders correctly 1`] = `
<div
  class="fd-alert fd-alert--warning fd-alert--dismissible"
  id="warningAlert"
  role="alert"
>
  <button
    aria-controls="alert"
    aria-label="Close"
    class="fd-alert__close"
  />
   
</div>
`;

exports[`Alert renders correctly 2`] = `
<div
  class="fd-alert fd-alert--error"
  id="errorAlert"
  role="alert"
>
  <!---->
   
</div>
`;
